src/vellum/workflows/workflows/base.py

```diff
@@ -255,9 +256,41 @@ def stream(
             external_inputs=external_inputs,
             cancel_signal=cancel_signal,
         ).stream():
-            if WorkflowEventType(event.name.split(".")[0].upper()) in event_types:
+            if should_yield(event):
                 yield event
 
+    def DEFAULT_EVENT_FILTER(self, event: WorkflowEvent) -> bool:
+        if (
+            event.name == "workflow.execution.initiated"
+            or event.name == "workflow.execution.resumed"
+            or event.name == "workflow.execution.fulfilled"
+            or event.name == "workflow.execution.rejected"
+            or event.name == "workflow.execution.paused"
+            or event.name == "workflow.execution.streaming"
+        ):
+            return event.workflow_definition == self.__class__
+
+        return False
+
+    def ROOT_EVENT_FILTER(self, event: WorkflowEvent) -> bool:
```

Comment:
the naming is weird, but I wanted to denote "public, easy to use filters for common use cases". I chose constant naming (all caps) for this but am open to alternatives